googletest/include/gtest/internal/gtest-port.h

```diff
@@ -1331,6 +1335,9 @@ class ThreadWithParam : public ThreadWithParamBase {
   void Join() {
     if (!finished_) {
       GTEST_CHECK_POSIX_SUCCESS_(pthread_join(thread_, nullptr));
+      // Yield to the joining thread to ensure its destructors are run
+      // before marking the thread as finished.
+      sched_yield();
```

Comment:
Relying on sched_yield to get interleaving is almost never correct.  The joined thread should have its dtors run before join returns regardless.  I have serious concerns about this

Reply:
I agree that the call to sched_yield was the one arguable change in this PR.
The reason for adding it was that without it the TEST(ThreadLocalTest, DestroysManagedObjectAtThreadExit) test fails.
It might be a better idea to simply disable this test rather than to try to force it to work.

Rebased to master and removed the call to sched_yield and put the one failing test inside a #if !defined(GTEST_OS_GNU_HURD) instead.

Reply:
Could you say more about why the test doesn't work on this platform? At the very least we should have a comment explaining why the test cannot pass on this platform, and maybe some notes on workarounds if it affects users who need the thread local destructors to run. But ideally fixing the underlying issue is preferable.

Reply:
When I run the test on exodar.debian.net it fails every time:
```
$ googletest/googletest-port-test
[ ... ]
[==========] 49 tests from 16 test suites ran. (5090 ms total)
[  PASSED  ] 48 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ThreadLocalTest.DestroysManagedObjectAtThreadExit
```
However, if I run only the thread-local test it succeeds:
```
$ googletest/googletest-port-test --gtest_filter=ThreadLocalTest.*
[ ... ]
[==========] 9 tests from 1 test suite ran. (20 ms total)
[  PASSED  ] 9 tests.
```
So there seems to be some interference with other tests.

Reply:
If I move the death tests to a separate file there is no more interference, and all tests succeed.
```
      Start 26: googletest-port-death-test
26/62 Test #26: googletest-port-death-test .................   Passed    7.34 sec
      Start 27: googletest-port-test
27/62 Test #27: googletest-port-test .......................   Passed    3.13 sec
```
I have removed the #ifndef disabling the DestroysManagedObjectAtThreadExit test and added a commit that does the above split. Is this acceptable?

PS. Can you address the memory usage during compilation? To build the package on the Debian hurd test machine, which had 3 GB of memory, I have to apply the proposed change from #3199 in order not to run out of memory.

Reply:
@ellert I can reproduce the failure on exodar, however on my VM all the tests work fine (master + the current patch #1 in this series). I'd say there might be something else going on, specific to Hurd, and to discuss it in a Hurd list (bug-hurd upstream, for example).

Let's go ahead with the plain GNU/Hurd enablement, and in case fixes can be applied later on. WDYT?

Reply:
@pinotree Thank you for redoing the test on exodar and confirming my findings. If the issue with the interference is not a general problem that happens for all hurd installations, I agree that maybe we can ignore it for now and only apply the first commit of the PR. It would be interesting to know why it happens on exodar though, and also if it happens on the Debian build servers. If it does happen on the build servers I guess the second commit of the PR can be added as a patch to the Debian source package until another solution is found.

I have removed the second commit from the PR.